### PR TITLE
Feature/zen 18634

### PR DESCRIPTION
**Pull request by burbel**

### CODE REVIEW

File: src/zenoss/toolbox/zencatalogscan.py [hunk 1]
```diff
@@ -187,14 +187,35 @@ def global_catalog_rids(catalog_name, catalog_list, fix, max_cycles, dmd, log):
             log.debug("Calling transaction.abort() to minimize memory footprint")
             transaction.abort()
 
-
-
-def scan_catalog(catalog_name, catalog_list, fix, max_cycles, dmd, log):
+    if create_events:
+        if number_of_issues > 0:
+            eventSummaryMsg = "\'%s\' - %d Error(s) Detected (%d total items)" % \
```
Comment (jpeacock-zenoss) [inline src/zenoss/toolbox/zencatalogscan.py]: The backslashes are not necessary here.  Single quotes inside double quotes work without backslashery.

File: src/zenoss/toolbox/zencatalogscan.py [hunk 2]
```diff
@@ -187,14 +187,35 @@ def global_catalog_rids(catalog_name, catalog_list, fix, max_cycles, dmd, log):
...
+                                  ('global_catalog_RIDs', number_of_issues, number_of_items)
+            eventSeverity = 4
+            eventDetails = "Consult https://support.zenoss.com/hc/en-us/articles/203118075 for additional information and resolution options"
+        else:
+            eventSummaryMsg = "\'%s\' - No Errors Detected (%d total items)" % \
+                                  ('global_catalog_RIDs', number_of_items)
+            eventSeverity = 1
+            eventDetails = "Consult https://support.zenoss.com/hc/en-us/articles/203118075 for additional information"
```
Comment (jpeacock-zenoss) [inline src/zenoss/toolbox/zencatalogscan.py]: Most of the string assigned to `eventDetails` is redundant.
You could assign shorter string prior the `if` statement then append the additional bits when needed.  That way you're recording the same information in your code twice.

File: src/zenoss/toolbox/zencatalogscan.py [hunk 3]
```diff
@@ -187,14 +187,35 @@ def global_catalog_rids(catalog_name, catalog_list, fix, max_cycles, dmd, log):
...
+
+        dmd.ZenEventManager.sendEvent({
+                'device'     : 'localhost',
+                'summary'    : eventSummaryMsg,
+                'component'  : 'zencatalogscan',
+                'severity'   : eventSeverity,
+                'eventClass' : '/Status',
+                'eventKey'   : "zencatalogscan_global_catalog_RIDs",
+                'details'    : eventDetails
+            })
```
Comment (jpeacock-zenoss) [inline src/zenoss/toolbox/zencatalogscan.py]: Minor formatting: de-indent lines 203-210 back by one indent (four spaces)

File: src/zenoss/toolbox/zencatalogscan.py [hunk 4]
```diff
@@ -251,9 +272,27 @@ def scan_catalog(catalog_name, catalog_list, fix, max_cycles, dmd, log):
...
-    return False
+    if create_events:
+        if number_of_issues > 0:
+            eventSummaryMsg = "\'%s\' - %d Error(s) Detected (%d total items)" % (catalog_name, number_of_issues, initial_catalog_size)
+            eventSeverity = 4  
+            eventDetails = "Consult https://support.zenoss.com/hc/en-us/articles/203118075 for additional information and resolution options"
+        else:
+            eventSummaryMsg = "\'%s\' - No Errors Detected (%d total items)" % (catalog_name, initial_catalog_size)
+            eventSeverity = 1
+            eventDetails = "Consult https://support.zenoss.com/hc/en-us/articles/203118075 for additional information"
```
Comment (jpeacock-zenoss) [inline src/zenoss/toolbox/zencatalogscan.py]: Same comment here about the backslashes and the redundant info used for the event details message.

File: src/zenoss/toolbox/zencatalogscan.py [hunk 5]
```diff
@@ -251,9 +272,27 @@ def scan_catalog(catalog_name, catalog_list, fix, max_cycles, dmd, log):
...
+     
+        dmd.ZenEventManager.sendEvent({ 
+                'device'     : 'localhost',
+                'summary'    : eventSummaryMsg,
+                'component'  : 'zencatalogscan',
+                'severity'   : eventSeverity,
+                'eventClass' : '/Status',
+                'eventKey'   : "zencatalogscan_%s" % (catalog_name),
+                'details'    : eventDetails
+            })
```
Comment (jpeacock-zenoss) [inline src/zenoss/toolbox/zencatalogscan.py]: formatting.  back it up four spaces.

File: src/zenoss/toolbox/zencatalogscan.py (hunk 3 above)
Comment (dougsyer) [inline src/zenoss/toolbox/zencatalogscan.py]: can you give this an eventClassKey?

Comment (burbel) [inline src/zenoss/toolbox/zencatalogscan.py]: Doug, I have set an eventClassKey using the same value as 'eventKey' - would you prefer a different value for eventClassKey?

**Pull request merged**